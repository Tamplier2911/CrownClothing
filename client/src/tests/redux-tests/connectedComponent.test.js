import { shallow } from "enzyme";
import React from "react";
import ConnectedComponent from "./connectedComponent";

// using manual store mocking or
// https://github.com/dmitry-zaets/redux-mock-store

// all enzme methods
// https://airbnb.io/enzyme/docs/api/shallow.html

// beforeEach(() =>{})

// wrapper.instance() - gives access to all methods
// expect(wrapper.instance().methodName([])).toEqual([])

// TESTING CONNECTED COMPONENTS
describe("<ConnectedComponent />", () => {
  it("expect to render connected component", () => {
    const mockStore = {
      getState() {
        return {
          user: {},
          cartDropdown: [],
          directory: [],
          shop: {}
        };
      }
    };
    const wrapper = shallow(<ConnectedComponent store={mockStore} />);
    expect(wrapper).toMatchSnapshot();
  });
});
