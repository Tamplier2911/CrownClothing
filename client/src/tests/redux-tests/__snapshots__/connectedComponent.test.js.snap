// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectedComponent /> expect to render connected component 1`] = `ShallowWrapper {}`;
